openedx/core/djangoapps/notifications/tasks.py

```diff
@@ -98,3 +107,32 @@ def send_notifications(user_ids, course_key, app_name, notification_type, contex
             ))
     # send notification to users but use bulk_create
     Notification.objects.bulk_create(notifications)
+
+
+def update_user_preference(preference: CourseNotificationPreference, user, course_id):
+    """
+    Update user preference if config version is changed.
+    """
+    ver = get_course_notification_preference_config_version()
+    if preference.config_version != ver:
+        return preference.get_updated_user_course_preferences(user, course_id)
+    return preference
+
+
+def create_notification_pref_if_not_exists(user_ids, preferences):
+    """
+    Create notification preference if not exist.
+    """
+    new_preferences = []
+
+    for user_id in user_ids:
+        if not any(preference.user_id == user_id for preference in preferences):
+            new_preferences.append(CourseNotificationPreference(
+                user_id=user_id,
+                course_id=preferences[0].course_id,
+            ))
+            logger.info('Creating new notification preference for user because it does not exist.')
+    if new_preferences:
+        CourseNotificationPreference.objects.bulk_create(new_preferences, ignore_conflicts=True)
```

Comment:
Why do we need ignore_conflicts here?

Reply:
added comment in code with explaination.